### PR TITLE
Merge PyUtils into ITKCommon module

### PR DESCRIPTION
**Pull request by dzenanz**
There is no need for this to be a special module,
which complicates its build setup. Closes #5027.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Core/Common/CMakeLists.txt [hunk 1]
```diff
@@ -131,6 +131,7 @@ set(ITK_USE_TBB ${Module_ITKTBB})
 configure_file(src/itkConfigure.h.in itkConfigure.h)
 configure_file(src/itkConfigurePrivate.h.in itkConfigurePrivate.h)
 
+set(ITKCommon_SYSTEM_INCLUDE_DIRS "${Python3_INCLUDE_DIRS}")
```
[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/Core/Common/CMakeLists.txt:134]: I tried bridge module first, but I don't know how to make all the modules dependent on it when wrapping is enabled.

[the remark this answers is not in the document]
Comment (thewtex) [reply Modules/Core/Common/CMakeLists.txt:134]: How about when wrapping is enabled `ITKCommon` depends on `ITKPyUtils` in ITKCommon's *itk-module.cmake*?

[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/Core/Common/CMakeLists.txt:134]: That causes circular dependency ITKCommon  -> ITKPyUtils -> ITKCommon.

[the remark this answers is not in the document]
Comment (thewtex) [reply Modules/Core/Common/CMakeLists.txt:134]: What other modules need to depend on ITKPyUtils? Shouldn't ITKPyUtils depend on ITKCommon?

[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/Core/Common/CMakeLists.txt:134]: Every module with name ending in Python wants it. Dependency should be `ITKVoronoiPython` -> `ITKPyUtils` -> `ITKCommon`, for all `ITK*Python` targets.

[the remark this answers is not in the document]
Comment (thewtex) [reply Modules/Core/Common/CMakeLists.txt:134]: In that case, the configuration will In that case, the configuration must be updated to exclude the requirement for *Python.h* when wrapping is not enabled.need to be updated to not depend on the presence of *Python.h* when the wrapping is not enabled.

[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/Core/Common/CMakeLists.txt:134]: PyCommand is excluded from building when wrapping is disabled. The latest force push should take care of auto-generated header tests.

**Comment by dzenanz** [comment]
The latest force-push is a plan rebase on master.

File: Modules/Core/Common/CMakeLists.txt [hunk 2]
```diff
@@ -131,6 +131,7 @@ set(ITK_USE_TBB ${Module_ITKTBB})
 configure_file(src/itkConfigure.h.in itkConfigure.h)
 configure_file(src/itkConfigurePrivate.h.in itkConfigurePrivate.h)
```
Comment (thewtex) [inline Modules/Core/Common/CMakeLists.txt:133]: `if(ITK_WRAP_PYTHON)`

**Review by thewtex — CHANGES_REQUESTED** [review]

File: Modules/Core/Common/CMakeLists.txt (hunk 2 above)
Comment (dzenanz) [reply Modules/Core/Common/CMakeLists.txt:133]: Latest force-push adds this condition. I did not put it there initially as I thought that variable would be empty when wrapping is off. But I now realize that it comes from finding Python, and it is usually found even when wrapping is off.

**Review by thewtex — APPROVED** [review]
@dzenanz thank you!!

**Pull request merged**